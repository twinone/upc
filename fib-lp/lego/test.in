Grid 10 20
B1 = PLACE (2,2) AT (1,1)
B2 = PLACE (2,4) AT (3,1)
MOVE B2 NORTH 1
MOVE B2 EAST 5
B2 = B1 PUSH B2
B3 = (2,3) PUSH B2
B2 = (2,3) PUSH B1 POP B2

WHILE ( FITS (B2, 2, 1, 2)) [
 B2 = (2,1) PUSH B2
]

HEIGHT(B2)

B4 = PLACE (2,2) AT (5,5)

TOWER10

DEF TOWER10
WHILE (HEIGHT(B4) > 1 AND HEIGHT(B4) < 10) [
 B4 = (2,2) PUSH B4
]
ENDEF

