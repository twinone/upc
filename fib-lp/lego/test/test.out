============================== AST ==============================
 list
  \__(Grid)
  |      \__(10)
  |      \__(10)
  \__list
  |      \__(=)
  |      |   \__(B1)
  |      |   \__(PLACE)
  |      |           \__list
  |      |           |      \__(2)
  |      |           |      \__(2)
  |      |           \__list
  |      |                  \__(1)
  |      |                  \__(1)
  |      \__(=)
  |      |   \__(B2)
  |      |   \__(PLACE)
  |      |           \__list
  |      |           |      \__(2)
  |      |           |      \__(4)
  |      |           \__list
  |      |                  \__(3)
  |      |                  \__(1)
  |      \__(MOVE)
  |      |      \__(B2)
  |      |      \__(NORTH)
  |      |      \__(1)
  |      \__(MOVE)
  |      |      \__(B2)
  |      |      \__(EAST)
  |      |      \__(5)
  |      \__(=)
  |      |   \__(B2)
  |      |   \__(PUSH)
  |      |          \__(B1)
  |      |          \__(B2)
  |      \__(=)
  |      |   \__(B3)
  |      |   \__(PUSH)
  |      |          \__list
  |      |          |      \__(2)
  |      |          |      \__(3)
  |      |          \__(B2)
  |      \__(=)
  |      |   \__(B2)
  |      |   \__(PUSH)
  |      |          \__list
  |      |          |      \__(2)
  |      |          |      \__(3)
  |      |          \__(POP)
  |      |                \__(B1)
  |      |                \__(B2)
  |      \__(WHILE)
  |      |       \__(FITS)
  |      |       |      \__(B2)
  |      |       |      \__(2)
  |      |       |      \__(1)
  |      |       |      \__(2)
  |      |       \__list
  |      |              \__(=)
  |      |                  \__(B2)
  |      |                  \__(PUSH)
  |      |                         \__list
  |      |                         |      \__(2)
  |      |                         |      \__(1)
  |      |                         \__(B2)
  |      \__(HEIGHT)
  |      |        \__(B2)
  |      \__(=)
  |      |   \__(B4)
  |      |   \__(PLACE)
  |      |           \__list
  |      |           |      \__(2)
  |      |           |      \__(2)
  |      |           \__list
  |      |                  \__(5)
  |      |                  \__(5)
  |      \__(TOWER10)
  \__list
         \__(DEF)
               \__(TOWER10)
               \__list
                      \__(WHILE)
                              \__(AND)
                              |     \__(>)
                              |     |   \__(HEIGHT)
                              |     |   |        \__(B4)
                              |     |   \__(0)
                              |     \__(<)
                              |         \__(HEIGHT)
                              |         |        \__(B4)
                              |         \__(10)
                              \__list
                                     \__(=)
                                         \__(B4)
                                         \__(PUSH)
                                                \__list
                                                |      \__(2)
                                                |      \__(2)
                                                \__(B4)
============================== GRID =============================
  0   0   0   0   0   0   0   2   2   0 
  0   0   0   0   0   0   0   2   2   0 
  0   0   0   0   0   0   0   2   2   0 
  0   0   0   0   0   0   0   2   2   0 
  0   0   0   0  10  10   0   0   0   0 
  0   0   0   0  10  10   0   0   0   0 
  0   0   0   0   0   0   0   0   0   0 
  0   0   0   0   0   0   0   0   0   0 
  0   0   0   0   0   0   0   0   0   0 
  0   0   0   0   0   0   0   0   0   0 
=================================================================
